Mathlib/Order/Prod/Lex/Monoid.lean

```diff
@@ -0,0 +1,125 @@
+/-
+Copyright (c) 2025 Yakov Pechersky. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yakov Pechersky
+-/
+import Mathlib.Algebra.Group.Prod
+import Mathlib.Algebra.Order.Hom.Monoid
+import Mathlib.Data.Prod.Lex
+import Mathlib.Order.Prod.Lex.Hom
+
+/-!
+# Order homomorphisms for products of ordered monoids
+
+This file defines order homomorphisms for products of ordered monoids, for both the plain product
+and the lexicographic product.
+
+The product of ordered monoids `α × β` is an ordered monoid itself with both natural inclusions
+and projections, making it the coproduct as well.
+
+TODO: Create the "OrdCommMon" category.
+
+-/
+
+namespace OrderMonoidHom
+
+variable (α β : Type*) [OrderedCommMonoid α] [OrderedCommMonoid β]
+
+/-- Given ordered monoids M, N, the natural inclusion ordered homomorphism from M to M × N. -/
+@[to_additive (attr := simps!) "Given ordered additive monoids M, N, the natural inclusion ordered
+homomorphism from M to M × N."]
+def inl : α →*o α × β where
+  __ := MonoidHom.inl _ _
+  monotone' _ _ := by simp
+
+/-- Given ordered monoids M, N, the natural inclusion ordered homomorphism from N to M × N. -/
+@[to_additive (attr := simps!) "Given ordered additive monoids M, N, the natural inclusion ordered
+homomorphism from N to M × N."]
+def inr : β →*o α × β where
+  __ := MonoidHom.inr _ _
+  monotone' _ _ := by simp
+
+/-- Given ordered monoids M, N, the natural projection ordered homomorphism from M × N to M. -/
+@[to_additive (attr := simps!) "Given ordered additive monoids M, N, the natural projection ordered
+homomorphism from M × N to M."]
+def fst : α × β →*o α where
+  __ := MonoidHom.fst _ _
+  monotone' _ _ := by simp +contextual [Prod.le_def]
```

Comment:
```suggestion
  monotone' := monotone_fst
```
etc. I think every proof in this file probably should have a single lemma you can invoke.